cc/crypto/hpke/recipient_context.h

```diff
@@ -35,55 +35,43 @@ struct KeyPair {
   static absl::StatusOr<KeyPair> Generate();
 };
 
-class RecipientRequestContext {
+class RecipientContext {
  public:
-  RecipientRequestContext(std::unique_ptr<EVP_AEAD_CTX> aead_request_context,
-                          std::vector<uint8_t> request_nonce)
-      : aead_request_context_(std::move(aead_request_context)),
-        request_base_nonce_(request_nonce),
-        sequence_number_(0) {}
+  RecipientContext(std::unique_ptr<EVP_AEAD_CTX> request_aead_context,
+                   std::vector<uint8_t> request_base_nonce, uint64_t request_sequence_number,
+                   std::unique_ptr<EVP_AEAD_CTX> response_aead_context,
+                   std::vector<uint8_t> response_base_nonce, uint64_t response_sequence_number)
+      : request_aead_context_(std::move(request_aead_context)),
+        request_base_nonce_(request_base_nonce),
+        request_sequence_number_(request_sequence_number),
+        response_aead_context_(std::move(response_aead_context)),
+        response_base_nonce_(response_base_nonce),
+        response_sequence_number_(response_sequence_number) {}
```

Comment:
Having a constructor with this many parameters can make things confusing. In order to clear this up in the code could you either:
1) Create a struct to encompass some of the arguments. For example ```struct RequestContext``` which has all the request based parameters and ```struct ResponseContext``` which has all the response based parameters.
OR
2) Each call to this constructor you comment what the variable is being assigned to. For example: https://github.com/project-oak/oak/blob/main/cc/crypto/hpke/sender_context.cc#L101-L112

Reply:
The amount of parameters would be reduced to 2 once we move to random nonces, so will add comments for each argument for now